File: targets/numworks/tests/camlboot/benchs/apply2.ml
let double f x = f (f x) ;;
let quad n = double double n ;;
let oct n = quad quad n  ;;
let succ n = n + 1 ;;

let repeat n =
  for i = 0 to n do
    let _ = double (quad succ) n in ()
  done

let () =
  let start = millis () in
  repeat 1000; (* x 1000 to compare with OMicroB *)
  let stop = millis () in
  print_int (stop-start)

File: targets/numworks/tests/camlboot/benchs/apply2.py
import time
# from microbit import *

double = lambda f : lambda x : f (f(x))
quad = lambda n : double(double)(n)
oct = lambda n : quad(quad)(n)
succ = lambda n : n + 1

def test_apply(n):
    i = 0
    r = 0
    while (i < n):
        r = r + double(quad(succ))(n-i )
        i = i + 1
    print(r)
    return r

start = time.monotonic()
test_apply(1000)
stop = time.monotonic()
print(stop-start)
